api/dataprocessor.go

```diff
@@ -322,36 +321,38 @@ func AggMetricKey(key, archive string, aggSpan uint32) string {
 	return fmt.Sprintf("%s_%s_%d", key, archive, aggSpan)
 }
 
-func (s *Server) getSeriesFixed(req models.Req, consolidator consolidation.Consolidator) []schema.Point {
-	ctx := newRequestContext(&req, consolidator)
-	res := s.getSeries(ctx)
-	res.Points = append(s.itersToPoints(ctx, res.Iters), res.Points...)
+func (s *Server) getSeriesFixed(ctx context.Context, req models.Req, consolidator consolidation.Consolidator) []schema.Point {
+	rctx := newRequestContext(&req, consolidator)
```

Comment:
i think it might be nicer to just make `ctx` an attribute of `rctx`, otherwise having two contexts around can be confusing.